packages/fx-core/src/client/teamsDevPortalClient.ts

```diff
@@ -79,33 +79,37 @@ export class RetryHandler {
   }
 }
 
-class TeamsDevPortalClient {
-  globalEndpoint: string;
+export class TeamsDevPortalClient {
   regionEndpoint?: string;
-  constructor() {
+
+  getGlobalEndpoint(): string {
     if (process.env.APP_STUDIO_ENV && process.env.APP_STUDIO_ENV === "int") {
-      this.globalEndpoint = "https://dev-int.teams.microsoft.com";
+      return "https://dev-int.teams.microsoft.com";
     } else {
-      this.globalEndpoint = "https://dev.teams.microsoft.com";
+      return "https://dev.teams.microsoft.com";
     }
   }
 
-  setRegionEndpoint(regionEndpoint: string) {
+  setRegionEndpoint(regionEndpoint: string): void {
     this.regionEndpoint = regionEndpoint;
   }
 
-  async setRegionEndpointByToken(regionToken: string) {
+  async setRegionEndpointByToken(authSvcToken: string): Promise<void> {
+    if (this.getGlobalEndpoint() === "https://dev-int.teams.microsoft.com") {
+      // Do not set region for INT env
```

Comment:
Why don't we set the region for INT env?

Reply:
Saw the logic, please ignore.

Reply:
Be consistant to the original logic. Maybe INT ent has no region endpoint.